### PR TITLE
v2v-ci various changes

### PR DESCRIPTION
**Pull request by istein1**
Merge istein1/v2v-ci recent changes into rhevm-qe-automation/v2v-ci

### CODE REVIEW

File: roles/v2v.ci/tasks/main.yml [hunk 1]
```diff
@@ -22,6 +22,12 @@
   tags:
     - miq_create_vms
 
+- import_tasks: remove_rhv_vms.yml
+  delegate_to: localhost
+  connection: local
```
Comment (kobihk) [inline roles/v2v.ci/tasks/main.yml]: delegate_to: localhost
  connection: local
I think the connection will be enough no need both

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml [hunk 2]
```diff
@@ -9,47 +9,77 @@
...
+  - name: Extract migration plan ID from JSON
     set_fact:
-      infra_mapping_id: "{{ infra_mapping_json.json | json_query(query) | first }}"
+      migration_plan_id: "{{ migration_plan_json.json | json_query(query) | first }}"
 
-  - name: Collect target VM ids
+  - name: Remove migration plan if already exists
     uri:
-      url: "https://{{ inventory_hostname }}/api/vms/?filter[]=name={{ item | regex_replace (' ', '%20') }}&expand=resources&attributes=name"
+      url: "https://{{ inventory_hostname }}/api/service_templates/{{migration_plan_id}}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_config_migration_plan.yml]: please keep the convention to add space before and after the curly brackets like:
      url: "https://{{ inventory_hostname }}/api/service_templates/{{ migration_plan_id }}"

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml [hunk 3]
```diff
@@ -9,47 +9,77 @@
...
-      body: "{{ lookup('template', 'templates/migration_plan.j2') | from_yaml }}"
-  when: migration_plan_json.json.subcount == 0
+- name: Extract infra mapping ID from JSON
+  set_fact:
+    infra_mapping_id: "{{ infra_mapping_json.json | json_query(query) | first }}"
+
+- name: Collect target VM ids
+  uri:
+    url: "https://{{ inventory_hostname }}/api/vms/?filter[]=name={{ item | regex_replace (' ', '%20') }}&expand=resources&attributes=name&filter[]=vendor=vmware&filter[]=ems_id!=null"
+    user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_config_migration_plan.yml]: do we have a file with default values that we can set it there?

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml [hunk 4]
```diff
@@ -9,47 +9,77 @@
...
+    password: "{{ miq_app_password | default('smartvm') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_config_migration_plan.yml]: same ^^

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml [hunk 5]
```diff
@@ -9,47 +9,77 @@
...
+    vm_query: "[*].json.resources[*].id"
+
+- name: Extract target VM ids from JSON
+  set_fact:
+    vm_ids: "{{ vm_ids_json.results | json_query(vm_query) }}"
+
+- name: Remove migrated tag from VMs
+  uri:
+    url: "https://{{ inventory_hostname }}/api/vms/{{ item }}/tags"
+    user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_config_migration_plan.yml]: if you set it in the default file you can save the repeat of default(xxx) here and in the line below

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml [hunk 6]
```diff
@@ -9,47 +9,77 @@
...
+      action: unassign
+      resources:
+        - category: "transformation_status"
+          name: "migrated"
+  with_items: "{{ vm_ids }}"
+
+- name: Create migration plan
+  uri:
+    url: "https://{{ inventory_hostname }}/api/service_templates"
+    user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_config_migration_plan.yml]: same ^^

File: roles/v2v.ci/tasks/miq_config_ovirt_conversion_hosts.yml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 - block:
     - name: Collect the list of host authentications
       uri:
-        url: "https://{{ inventory_hostname }}/api/hosts/?filter[]=name={{ item }}&expand=resources&attributes=authentications"
+        url: "https://{{ inventory_hostname }}/api/hosts/?filter[]=name={{ hostvars[item].rhv_host_name }}&expand=resources&attributes=authentications"
         user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_config_ovirt_conversion_hosts.yml]: same ^^

File: roles/v2v.ci/tasks/miq_config_vmware_esx_hosts.yml [hunk 8]
```diff
@@ -2,7 +2,7 @@
 - block:
     - name: Collect the list of host authentications
       uri:
-        url: "https://{{ inventory_hostname }}/api/hosts/?filter[]=name={{ item }}&expand=resources&attributes=authentications"
+        url: "https://{{ inventory_hostname }}/api/hosts/?filter[]=name={{ hostvars[item].vmw_host_name }}&expand=resources&attributes=authentications"
         user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_config_vmware_esx_hosts.yml]: same ^^

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml [hunk 9]
```diff
@@ -0,0 +1,126 @@
+---
+- block:
+    - name: Find host JSON
+      uri:
+        url: "https://{{ inventory_hostname }}/api/hosts/?filter[]=name={{ hostvars[item].rhv_host_name }}&expand=resources&attributes=authentications"
+        user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: same ^^

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+        query: "resources[?name=='{{ hostvars[item].rhv_host_name }}'].id"
+
+    - name: Extracting host id from JSON
+      set_fact:
+        host_id: "{{ host_json.json | json_query(query) | first }}"
+
+    - name: Look for conversion host {{ hostvars[item].rhv_host_name }}
+      uri:
+        url: "https://{{ inventory_hostname }}/api/conversion_hosts/?filter[]=name={{ hostvars[item].rhv_host_name }}&expand=resources"
+        user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: same ^^

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+            query: "resources[?name=='{{ hostvars[item].rhv_host_name }}'].id"
+
+        - name: Extract conversion host ID from JSON
+          set_fact:
+             conversion_host_id: "{{ conversion_host_json.json | json_query(query) | first }}"
+
+        - name: Remove conversion host {{ hostvars[item].rhv_host_name }}
+          uri:
+            url: "https://{{ inventory_hostname }}/api/conversion_hosts/{{conversion_host_id}}"
+            user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: same ^^

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+            password: "{{ miq_app_password | default('smartvm') }}"
+            method: DELETE
+            status_code: 204
+            validate_certs: no
+
+        - name: Wait until conversion host {{ hostvars[item].rhv_host_name }} is removed
+          uri:
+            url: "https://{{ inventory_hostname }}/api/conversion_hosts/?filter[]=name={{ hostvars[item].rhv_host_name }}&expand=resources"
+            user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: same ^^

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+      delegate_to: "{{ v2v_ci_rhv_hostname }}"
+      shell: tail -n +4 /etc/pki/ovirt-engine/keys/engine_id_rsa
+      register: ssh_key
+
+#    - debug: msg="{{ssh_key.stdout}}"
+
+    - name: Enable conversion host {{ hostvars[item].rhv_host_name }}
+      uri:
+        url: "https://{{ inventory_hostname }}/api/conversion_hosts"
+        user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: same ^^

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+              resource_type: "ManageIQ::Providers::Redhat::InfraManager::Host"
+              resource_id: "{{ host_id }}"
+              max_concurrent_tasks: "{{ v2v_max_concurrent_conversions }}"
+              auth_user: "root"
+              vmware_vddk_package_url: "{{ vmware_vddk_package_url }}"
+              conversion_host_ssh_private_key: "{{ ssh_key.stdout }}"
+              vmware_ssh_private_key: "{{ vmware_ssh_private_key }}"
+      register: enable_conversion_host_json
+
+ #   - debug: msg="{{enable_conversion_host_json.json.results[0].task_href}}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: do we need this comment?

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+
+    - name: Check enable conversion host task url is valid
+      uri:
+        url: "https://{{ inventory_hostname }}/api/tasks/?filter[]=id={{enable_conversion_host_json.json.results[0].task_id}}"
+        user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: same ^^

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+      register: task_json
+      until: task_json.json.subcount == 1
+      retries: 20
+      delay: 5
+
+
+    - name: Check enable conversion host task has ended
+      uri:
+        url: "{{enable_conversion_host_json.json.results[0].task_href}}"
+        user: "{{ miq_app_username | default('admin') }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: same ^^

File: roles/v2v.ci/tasks/miq_create_vms.yml [hunk 17]
```diff
@@ -24,6 +12,12 @@
     cluster: "{{ v2v_ci_source_cluster }}"
   with_items: "{{ v2v_ci_migration_plan_vms }}"
 
+- set_fact:
+    vms_per_host: "{{ '1' if v2v_ci_migration_plan_vms|length == 1 else v2v_ci_migration_plan_vms|length / groups['vmware_esx_hosts']|length }}"
+#    vms_per_host: "{{ v2v_ci_migration_plan_vms|length / groups['vmware_esx_hosts']|length }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_create_vms.yml]: do we need this comment line?

File: roles/v2v.ci/tasks/miq_start_monitoring.yml [hunk 18]
```diff
@@ -21,9 +21,28 @@
   with_items:
     -  "{{ groups['ovirt_conversion_hosts'] }}"
 
+#- name: Start Collecting Conversion hosts logs
+#  delegate_to: "{{ item }}"
+#  script: roles/v2v.ci/files/conv_host_start_log.sh "v2v_test"
+#  with_items:
+#    -  "{{ groups['ovirt_conversion_hosts'] }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/miq_start_monitoring.yml]: do we need this comment out code?

File: roles/v2v.ci/tasks/remove_rhv_vms.yml [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+  ovirt_vm:
+    auth:
+      username: "{{ v2v_ci_rhv_user }}"
+      password: "{{ v2v_ci_rhv_password }}"
+      hostname: "{{ v2v_ci_rhv_hostname }}"
+      insecure: yes
+    state: absent
+    name: "{{ item }}"
+    cluster: "{{ v2v_ci_target_cluster }}"
+  with_items: "{{ v2v_ci_migration_plan_vms }}"
```
Comment (kobihk) [inline roles/v2v.ci/tasks/remove_rhv_vms.yml]: missing new line

**Review by kobihk — COMMENTED** [review]
please try to look at your code maybe you can create xx_uri_task.yaml to do the same uri that you use multiple times and just call it with import_tasks instead of repeat this code.

File: roles/v2v.ci/tasks/main.yml (hunk 1 above)
Comment (istein1) [reply roles/v2v.ci/tasks/main.yml]: delegate_to was removed

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml (hunk 2 above)
Comment (istein1) [reply roles/v2v.ci/tasks/miq_config_migration_plan.yml]: fixed

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml (hunk 3 above)
Comment (istein1) [reply roles/v2v.ci/tasks/miq_config_migration_plan.yml]: I removed the default value, and the explicit user/password
miq_app_username / miq_app_password are defined in the source yaml file.

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml (hunk 5 above)
Comment (istein1) [reply roles/v2v.ci/tasks/miq_config_migration_plan.yml]: Fixed, same as above.

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml (hunk 4 above)
Comment (istein1) [reply roles/v2v.ci/tasks/miq_config_migration_plan.yml]: Fixed, same as above.

File: roles/v2v.ci/tasks/miq_config_migration_plan.yml (hunk 6 above)
Comment (istein1) [reply roles/v2v.ci/tasks/miq_config_migration_plan.yml]: Fixed, same as above.

File: roles/v2v.ci/tasks/miq_config_ovirt_conversion_hosts.yml (hunk 7 above)
Comment (istein1) [reply roles/v2v.ci/tasks/miq_config_ovirt_conversion_hosts.yml]: Fixed, same as above.

File: roles/v2v.ci/tasks/miq_config_vmware_esx_hosts.yml (hunk 8 above)
Comment (istein1) [reply roles/v2v.ci/tasks/miq_config_vmware_esx_hosts.yml]: Fixed, same as above.

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml (hunk 9 above)
Comment (istein1) [reply roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: Fixed, same as above.

File: roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml (hunk 10 above)
Comment (istein1) [reply roles/v2v.ci/tasks/miq_conversion_hosts_enable.yml]: Fixed, same as above.

File: roles/v2v.ci/tasks/main.yml (hunk 1 above)
Comment (istein1) [reply roles/v2v.ci/tasks/main.yml]: I returned the delegate_to: localhost, after it failed on not having the correct ovirt-sdk version:
https://rhv-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/v2v-pipeline-prod/100
Here's the verification of it, after it was added back:
https://rhv-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/v2v-pipeline-prod/101

Comment (kobihk) [reply roles/v2v.ci/tasks/main.yml]: please try to remove the "connection: local" instead

**Review by kobihk — APPROVED** [review]

File: roles/v2v.ci/tasks/main.yml (hunk 1 above)
Comment (istein1) [reply roles/v2v.ci/tasks/main.yml]: I removed "connection: local"  from several tasks, and run a full v2v migration flow, that passed.

**Review by kobihk — APPROVED** [review]

**Pull request merged**